packages/plugin-commands-init/src/init.ts

```diff
@@ -34,8 +34,9 @@ export async function handler (
     })
   }
   // Using cwd instead of the dir option because the dir option
-  // is set to the first parent directory that has a package.json file.
-  const manifestPath = path.join(process.cwd(), 'package.json')
+  // is set to the first parent directory that has a package.json file
+  // But --dir option from cliOptions should be respected.
+  const manifestPath = path.join(opts.cliOptions.dir ?? process.cwd(), 'package.json')
```

Comment:
I believe you can do simply this
```suggestion
  const manifestPath = path.join(opts.dir, 'package.json')
```

Reply:
From the previous comment. The `opts.dir` 
> is set to the first parent directory that has a package.json file

So at this point, once the parent directory has `package.json` file, the init will fail.

But the `opts.cliOptions.dir` will effect only when the `--dir` option is specified